File: arrow.clj
;; Declare functions
(defn repeatChar [character nTimes]
  (apply str (repeat nTimes character)))

(defn drawSlice [floor height]
  (str (repeatChar " "(- (- height floor) 1)) 
    "*"
    (repeatChar " "(+ (* floor 2) 1))
    "*"))

;; Get arrow height
(println "What height the arrow should be ?")
(def height (Integer/parseInt (read-line)))

;; Draw the first point
(println (repeatChar " " (- height 1)) "*")

;; Draw the slices
(doseq [x (range (- height 1))]
  (println (drawSlice x height)))

;; Draw the base
(println (repeatChar "*" (+ (* 2 height) 1)))

;; Draw the tail
(doseq [x (range height)]
  (println (repeatChar " " (- height 1)) "*"))

File: arrow.rb
puts "What is the arrow's height ?"

height = gets.to_i

space_between = -1

(1..height * 2).each do |line|
  text = ''

  if line == height
    (1..height * 2 - 1).each { text += '*' }
  elsif line < height
    (1..height - line).each { text += ' '}

    text += '*'

    if space_between > 0
      (1..space_between).each { text += ' ' }
    end

    text += '*' if line != 1
  else
    (1..height - 1).each { text += ' ' }
    text += '*'
  end

  space_between += 2
  puts text
end
